### PR TITLE
meta: add type inferring for not_null, not_false and create_vector from polyfills

### PR DESCRIPTION
**Pull request by i582**
Since type inference occurs before lazy types are resolved, 
this does not work entirely for function return types.

Perhaps we should add the ability to add a lazy type, which 
would say that when resolving types, need to remove a certain 
type, for example, `false` or `null`.

#977

### CODE REVIEW

File: src/types/typesmap.go [hunk 1]
```diff
@@ -81,6 +81,18 @@ func (m Map) Map(fn func(string) string) Map {
 	return NewMapFromMap(mapped)
 }
 
+// Filter returns a new types map with the types of m for which fn returns true.
+// The result type map is never marked as precise.
+func (m Map) Filter(fn func(string) bool) Map {
+	mapped := make(map[string]struct{}, len(m.m))
```
Comment (quasilyte) [inline src/types/typesmap.go:87]: s/map/filtered/?

Comment (i582) [reply src/types/typesmap.go]: Done,

**Comment by quasilyte** [comment]
Conflicts.

**Comment by i582** [comment]
> Conflicts.

Fixed.

**Comment by i582** [comment]
Oh no...

**Comment by quasilyte** [comment]
:sweat_smile:

**Pull request merged**